deps/chakrashim/src/jsrtisolateshim.cc

```diff
@@ -670,6 +671,26 @@ void IsolateShim::SetPromiseRejectCallback(v8::PromiseRejectCallback callback) {
                                    reinterpret_cast<void*>(callback));
 }
 
+void IsolateShim::RunMicrotasks() {
+  // Loop until we've handled all the tasks, including the ones
+  // added by these tasks
+  while (!this->microtaskQueue.empty()) {
+    std::vector<MicroTask> batch;
+    std::swap(batch, this->microtaskQueue);
+
+    for (unsigned int i = 0; i < batch.size(); ++i) {
+      JsValueRef notUsed;
+      if (jsrt::CallFunction(batch[i].task, &notUsed) != JsNoError) {
+        JsGetAndClearException(&notUsed);  // swallow any exception from task
+      }
+    }
+  }
+}
+
+void IsolateShim::QueueMicrotask(JsValueRef task) {
+  this->microtaskQueue.emplace_back(task);
```

Comment:
@kfarnung Would you mind double checking this is correct? I'm intending that this constructs the `MicroTask` in place in the vector

Reply:
Looks correct to me.